File: app/src/main/java/io/github/freewebmovement/igniter/ui/servers/data/ServerListDataSource.java
package io.github.freewebmovement.igniter.ui.servers.data;

import androidx.annotation.WorkerThread;

import java.util.List;

import io.github.freewebmovement.igniter.persistence.TrojanConfig;

public interface ServerListDataSource {
    @WorkerThread
    List<TrojanConfig> loadServerConfigList();
    @WorkerThread
    void deleteServerConfig(TrojanConfig config);
    @WorkerThread
    void saveServerConfig(TrojanConfig config);
    @WorkerThread
    void replaceServerConfigs(List<TrojanConfig> list);
}

File: app/src/main/java/io/github/freewebmovement/igniter/activities/servers/data/ServerListDataSource.kt
package io.github.freewebmovement.igniter.activities.servers.data

import androidx.annotation.WorkerThread
import io.github.freewebmovement.igniter.persistence.TrojanConfig

interface ServerListDataSource {
    @WorkerThread
    fun loadServerConfigList(): List<TrojanConfig>

    @WorkerThread
    fun deleteServerConfig(config: TrojanConfig)

    @WorkerThread
    fun saveServerConfig(config: TrojanConfig)

    @WorkerThread
    fun replaceServerConfigs(list: List<TrojanConfig>)
}
